pages/announcement/AnnouncementHandler.php

```diff
@@ -78,13 +72,13 @@ public function view($args, $request)
         $this->setupTemplate($request);
 
         $announcementId = (int) array_shift($args);
-        $announcement = Repo::announcement()->get($announcementId);
+        $announcement = Announcement::find($announcementId);
         if (
             $announcement
-            && $announcement->getAssocType() == Application::getContextAssocType()
-            && $announcement->getAssocId() == $request->getContext()?->getId()
+            && $announcement->getAttribute('assocType') == Application::getContextAssocType()
+            && $announcement->getAttribute('assocId') == $request->getContext()?->getId()
             && (
-                $announcement->getDateExpire() == null || strtotime($announcement->getDateExpire()) > time()
+                $announcement->getAttribute('dateExpire') == null || strtotime($announcement->getAttribute('dateExpire')) > time()
```

Comment:
It should be possible to use [Eloquent date casts](https://laravel.com/docs/11.x/eloquent-mutators#date-casting) to get columns to come back automagically as Carbon objects; it shouldn't be necessary/desirable to use `strtotime` anymore.

Reply:
The case here is bit different here than the usual . we can have a cast define but as our types are coming from schema which will over write it in the `ModelWithSettings` trait's `convertSchemaToCasts` method where we are calling the `mergeCasts` which merge the provided casts with pre defined ones . I think if we want to apply some casts like 
```php
/**
 * Get the attributes that should be cast.
 *
 * @return array<string, string>
 */
protected function casts(): array
{
    return [
        'dateExpire' => 'datetime',
        'datePosted' => 'datetime',
    ];
}
```
we need to override the `Illuminate\Database\Eloquent\Concerns\HasAttributes::mergeCasts` to defined how it will handle and determine the precedence of provided casts over defined ones .

Reply:
Thinking about this a bit more, I think we should override the method `mergeCasts` inside the trait `ModelWithSettings` to allow model defined casts over write any casts coming form schema if schema available like 
```php
/**
     * Merge new casts with existing casts on the model.
     *
     * @param  array  $casts
     * @return $this
     */
    public function mergeCasts($casts)
    {
        $casts = $this->ensureCastsAreStringValues($casts);
        dump($this->casts);
        $this->casts = array_merge($casts, $this->casts);

        return $this;
    }
```
basically we are just altering how the array_merge getting done . 

As trait `ModelWithSettings` will defined directly in entity model classes, it's `mergeCasts` should have higher precedence of `HasAttributes::mergeCasts` used in `Illuminate\Database\Eloquent\Model` .